lib/utilities/legacyAWSBehavior.js

```diff
@@ -0,0 +1,24 @@
+import config from '../Config';
+
+/**
+ * check the legacy AWS behaviour
+ * For new configuration: check if locationConstraint has its
+ * legacyAwsBehavior put to true
+ * For old configuration: check if locationConstraint is 'us-east-1' and
+ * usEastBehavior is true
+ * @param {string | undefined} locationConstraint - value of user
+ * metadata location constraint header or bucket location constraint
+ * @param {boolean} usEastBehavior - only used for old configuration
+ * @return {boolean} - true if valid, false if not
+ */
+
+export default function isLegacyAwsBehavior(locationConstraint,
+  usEastBehavior) {
```

Comment:
if you don't want to send `usEastBehavior` as a parameter, you can also just use `config.usEastBehavior` in the code below, but it may be clearer what you are checking if you keep it as a parameter